core/search/index.py

```diff
@@ -228,8 +229,39 @@ def register_neural_search_fields(
             }
         )
 
-    def reindex(self) -> None:
-        helpers.reindex(self.client, self.name, self.name)
+    def reindex(self, fields: List[str]) -> None:
+        CHUNK_SIZE = 1000
+        THREAD_COUNT = 4
+
+        logger.info(f"Reindexing with {THREAD_COUNT} threads...")
+
+        def _generator() -> Generator[Dict[str, Any], Any, None]:
+            """Generator function to fetch all documents from the specified index."""
+            for hit in helpers.scan(self.client, index=self.name):
+                doc = {k: hit["_source"][k] for k in fields if k in hit["_source"]}
+
+                if not doc:
+                    continue
+
+                yield {
+                    "_op_type": "update",
+                    "_index": hit["_index"],
+                    "_id": hit["_id"],
+                    "doc": {
+                        k: hit["_source"][k] for k in fields if k in hit["_source"]
+                    },
+                    "doc_as_upsert": True,
+                }
+
+        deque(
```

Comment:
It seems that the deque is used to capture the output in the bulk operation, but then it is not used again to check for errors or the  result. I'm curious to know why you opted for deque on this particular case.

Reply:
It's what elasticsearch recommends in their documentation: https://elasticsearch-py.readthedocs.io/en/7.x/helpers.html

Reply:
Alright, but I was thinking we should check for errors like in this example: 

https://github.com/opensearch-project/opensearch-py/blob/fca8bd3d0e7f42f2458e924c36ac6a135a6bc622/samples/bulk/bulk-ld.py#L55

Reply:
Ok I can add that!